src/prometheus/definition.go

```diff
@@ -701,3 +724,41 @@ func getRawEntityID(parentGroupLabel, groupLabel, entityID string, groups defini
 	}
 	return rawEntityID, nil
 }
+
+func WithLabelValueInMetricNameAndWithLabelValueMappedToMetricValue(rawMetricName, label, format, valueLabel string, valueMap map[string]int) definition.FetchFunc {
+	return func(groupLabel, entityID string, rawGroups definition.RawGroups) (definition.FetchedValue, error) {
+		fetchedValues := definition.FetchedValues{}
+		metricsRaw, err := definition.FromRaw(rawMetricName)(groupLabel, entityID, rawGroups)
+		if err != nil {
+			return nil, fmt.Errorf("fetching metric %q for entity %q in group %q: %w", rawMetricName, entityID, groupLabel, err)
+		}
+
+		metrics, ok := metricsRaw.([]Metric)
+		if !ok {
+			return nil, fmt.Errorf("incompatible metric type for %q. Expected: []Metric. Got: %T", rawMetricName, metricsRaw)
+		}
```

Comment:
Will this work if KSM only reports one condition?

Reply:
Good point, it will not. Though it is guaranteed that multiple will be returned as kubelet itself reports 4 standard ones.

Reply:
I added suport for both single metric and multiple metrics, though it's a bit annoying. I think eventually all metrics should be slices.